File: Python3/0657_robot_return_to_origin.py
# https://leetcode.cn/problems/robot-return-to-origin/

class Solution:
    def judgeCircle(self, moves: str) -> bool:
        h, v = 0, 0
        for ch in moves:
            if ch == "R":
                h += 1
            elif ch == "L":
                h -= 1
            elif ch == "D":
                v += 1
            else:
                v -= 1
        return h == 0 and v == 0

File: Cpp/0657_robot_return_to_origin.cpp
// https://leetcode.cn/problems/robot-return-to-origin/

#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    bool judgeCircle(string moves) {
        int h = 0, v = 0;
        for (const auto &ch: moves) {
            if (ch == 'R') h += 1;
            else if (ch == 'L') h -= 1;
            else if (ch == 'D') v += 1;
            else v -= 1;
        }
        return h == 0 && v == 0;
    }
};

int main() {
    Solution sol;
    std::string moves = "UD";
    cout << sol.judgeCircle(moves) << '\n';
    return 0;
}